ta/pkcs11/src/user_ta_header_defines.h

```diff
@@ -15,7 +15,12 @@
 					 TA_FLAG_INSTANCE_KEEP_ALIVE)
 
 #define TA_STACK_SIZE			(4 * 1024)
+
+#ifdef CFG_PKCS11_TA_HEAP_SIZE
```

Comment:
Why the `#ifdef` since `CFG_PKCS11_TA_HEAP_SIZE` has a default value in `mk/config.mk`? (I am suspecting a possible issue if the PKCS#11 TA had to be built outside the optee_os environment since `CFG_PKCS11_TA_HEAP_SIZE` is not exported; do we need something like what is done for `CFG_TEE_TA_LOG_LEVEL` [here](https://github.com/OP-TEE/optee_os/blob/3.14.0/ta/ta.mk#L41)?)

Reply:
I would prefer not to add TA specific switch in devkit generic _ta.mk_.
I think it is easier to treat `CFG_PKCS11_TA_HEAP_SIZE` (a value) with `#ifdef` pragma to handle case where it is not defined and don't add burden in mandatory configs, as other `CFG_PKCS11_TA_xxx` switch that don't really mandate a default value (since not defined means disabled) despite we set one in _mk/config.mk_.

Reply:
We already do the same with `CFG_PKCS11_TA_TOKEN_COUNT` in [_ta/pkcs11/src/pkcs11_token.c_](https://github.com/OP-TEE/optee_os/blob/3.14.0/ta/pkcs11/src/pkcs11_token.c#L28-L32).

Reply:
Right. How about moving the `CFG_PKCS11_TA*` definitions to `ta/pkcs11/sub.mk`? Then they would always get a default value, and the `#ifdef` could be removed. In addition it makes more sense to define them in the TA folder since they are used only by the TA.

Reply:
I through we didn't want to set `CFG_xxx` values from _sub.mk_ files.

Reply:
It is allowed, although rarely done I agree, because we prefer to use them only to list source files and sub-directories. But that's because there is often a separate `.mk` file where more complex things are done (such as `core.mk` etc.). From a `make` point of view it doesn't make a difference. `sub.mk` is even listed as a valid place to define `CFG_` flags in `mk/config.mk`, see [here](https://github.com/OP-TEE/optee_os/blob/3.14.0/mk/config.mk#L14).

Reply:
ok, great, thanks.
I'll update this P-R to move all `CFG_PKCS11_TA_*` switches into _ta/pkcs11/sub.mk_.